### PR TITLE
fixed DeprecationWarning

### LINKED ISSUE #727: DeprecationWarning from setuptools 67.5.0 and above
Hi,

If `setuptools` 67.5.0 and above is installed, I receive a `DeprecationWarning` when running the tests with errors and warnings enabled (tested with Python 3.10.10):

```
  File "lib\site-packages\sorl\__init__.py", line 1, in <module>
    from pkg_resources import get_distribution, DistributionNotFound
  File "lib\site-packages\pkg_resources\__init__.py", line 121, in <module>
    warnings.warn("pkg_resources is deprecated as an API", DeprecationWarning)
DeprecationWarning: pkg_resources is deprecated as an API
```

With `setuptools` 67.4.0 and below, there is no `DeprecationWarning`.

### PR DESCRIPTION
**Pull request by LarryMartell**
fixed DeprecationWarning

Fixes #727 

Removes support for Python 3.7.

### CODE REVIEW

**Comment by claudep** [comment]
As I commented in the previous PR, Python 3.7 is EOL, so it should be removed from the test matrix. The the `install-requires` will not be needed any more.

**Comment by LarryMartell** [comment]
So are you saying the issue is closed and this PR is not needed any more?

**Comment by claudep** [comment]
No, it's not. However, if 3.7 support is dropped, this will simplify this patch, as the try/except will not be necessary.

**Comment by LarryMartell** [comment]
I have a PR our to fix the tests in master. Shall I remove the 3.7 support in that branch?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -57,4 +57,4 @@ jobs:
     - name: Upload coverage
       uses: codecov/codecov-action@v3
       with:
-        name: Python ${{ matrix.python-version }}
+        name: Python ${{ matrix.python-version }}
```
Comment (claudep) [inline .github/workflows/test.yml:60]: This change is not useful, please revert.

File: setup.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
     url='https://github.com/jazzband/sorl-thumbnail',
     packages=find_packages(exclude=['tests', 'tests.*']),
     platforms='any',
-    python_requires='>=3.7',
+    python_requires='>=3.8',
```
Comment (claudep) [inline setup.py:16]: You can also remove the 3.7 classifier below.

File: sorl/__init__.py [hunk 3]
```diff
@@ -1,6 +1,5 @@
-from pkg_resources import get_distribution, DistributionNotFound
-
 try:
-    __version__ = get_distribution("sorl-thumbnail").version
-except DistributionNotFound:
+    import importlib.metadata as importlib_metadata
+    __version__ = importlib_metadata.version("sorl-thumbnail")
+except Exception:
```
Comment (claudep) [inline sorl/__init__.py:4]: Is there a reason to catch exceptions here? Did you read somewhere that it was needed?

**Review by claudep — CHANGES_REQUESTED** [review]
Thanks, you are on the right track!

File: .github/workflows/test.yml (hunk 1 above)
Comment (LarryMartell) [reply .github/workflows/test.yml]: Done

File: setup.py (hunk 2 above)
Comment (LarryMartell) [reply setup.py:16]: Done

File: sorl/__init__.py (hunk 3 above)
Comment (LarryMartell) [reply sorl/__init__.py]: No, was just being defensive. I've removed it.

**Comment by LarryMartell** [comment]
After making these small changes the tests are failing in my forked branch, but it's not clear why.

https://github.com/LarryMartell/sorl-thumbnail/actions/runs/5650549930/job/15307090121#step:10:458

**Comment by uri-rodberg** [comment]
@claudep This PR looks fine to me now. Do you still require changes or is it OK with you now? And why are the tests canceled and failing? I expected the tests to pass now.

**Comment by LarryMartell** [comment]
The tests were passing until I made some very small changes Claude
requested. It’s not clear to me at all why they’re failing now.

**Comment by LarryMartell** [comment]
The tests are very persnickety. The presence or absence of a blank line
seems to make them pass or fail.

**Comment by uri-rodberg** [comment]
> The tests are very persnickety. The presence or absence of a blank line seems to make them pass or fail. On Mon, Jul 24, 2023 at 8:37 PM Larry Martell ***@***.***> wrote:
> […](#)
> The tests were passing until I made some very small changes Claude requested. It’s not clear to me at all why they’re failing now.

@LarryMartell Can you please add a blank line at the end of `.github/workflows/test.yml`, and a blank line at the end of `sorl/__init__.py`, and then we will check if the tests pass?

**Comment by claudep** [comment]
I created #738 to only make the Python 3.7 drop (and let Larry as author), so we can then concentrate on the `version` issue only.

**Comment by claudep** [comment]
According to https://pypi.org/project/setuptools-scm/#retrieving-package-version-at-runtime, we should still catch `PackageNotFoundError` for cases where the package is not installed (typically during tests or dev).

**Comment by LarryMartell** [comment]
test.yml already had a blank line. Added one to __init__.py, tests still
failing.

‪On Mon, Jul 24, 2023 at 10:40 PM ‫אורי‬‎ ***@***.***> wrote:‬

> The tests are very persnickety. The presence or absence of a blank line
> seems to make them pass or fail. On Mon, Jul 24, 2023 at 8:37 PM Larry

**Comment by claudep** [comment]
Blank lines have nothing to do with the failure. Please read my comment above.

**Comment by LarryMartell** [comment]
Here is the error from the test:

sorl/__init__.py:8:1: W391 blank line at end of file

Which comment? About PackageNotFound? I added that.

**Comment by claudep** [comment]
Looks good. Try now to rebase on master, the only changed file should be `sorl/__init__.py`.

**Comment by LarryMartell** [comment]
Removed blank line, tests are passing now.

**Comment by uri-rodberg** [comment]
@LarryMartell Do you know how to rebase? This PR should be rebased on branch master. It looks to me it's still not rebased. Please read about rebasing and rebase all your commits on branch master.

**Comment by LarryMartell** [comment]
Yes, I know how to rebase. My 727.1 branch has been rebased with master.

**Comment by LarryMartell** [comment]
I have a fork and from github I sync my fork's master with the upstream, so
it should be the same. But in any case I followed your exact directions and
rebased.

The 5 files shown as modified have indeed all been modified in my branch.


‪On Tue, Jul 25, 2023 at 9:44 AM ‫אורי‬‎ ***@***.***> wrote:‬

**Comment by LarryMartell** [comment]
There are 2 things in this branch - fixing the deprecation warning and
removing python 3.7.

**Comment by LarryMartell** [comment]
Are there any issues with this PR that are preventing it from being accepted and merged?

**Comment by uri-rodberg** [comment]
@LarryMartell Please contact me by email. Your mailbox is full.

**Review by JanMalte — APPROVED** [review]

**Comment by uri-rodberg** [comment]
I'm going to create another PR instead of this PR. Closing this PR.

**Pull request closed without merging**